components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -0,0 +1,380 @@
+     *
+     * @param eventProperties   Event properties.
+     * @param userStoreManager  User store manager.
+     * @return User.
+     */
+    private User getUser(Map eventProperties, UserStoreManager userStoreManager) {
+
+        String userName = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+        String tenantDomain = (String) eventProperties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+        String domainName = userStoreManager.getRealmConfiguration().
+                getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+
+        User user = new User();
+        user.setUserName(userName);
+        user.setTenantDomain(tenantDomain);
+        user.setUserStoreDomain(domainName);
+        return user;
+    }
+
+    /**
+     * If the mobile claim is updated, set it to the 'MOBILE_NUMBER_PENDING_VALUE_CLAIM' claim.
+     * Set thread local state to skip sending verification notification in inapplicable claim update scenarios.
+     *
+     * @param claims            Map of claims to be updated.
+     * @param userStoreManager  User store manager.
+     * @param user              User.
+     * @throws IdentityEventException
+     */
+    private void preSetUserClaimOnMobileNumberUpdate(Map<String, String> claims, UserStoreManager userStoreManager,
+                                                     User user) throws IdentityEventException {
+
+        if (IdentityRecoveryConstants.SkipMobileNumberVerificationOnUpdateStates.SKIP_ON_CONFIRM.toString().equals
+                (Utils.getThreadLocalToSkipSendingSmsOtpVerificationOnUpdate())) {
+            // Not required to handle in this handler.
+            return;
+        }
+
+        if (Utils.getThreadLocalToSkipSendingSmsOtpVerificationOnUpdate() != null) {
+            Utils.unsetThreadLocalToSkipSendingSmsOtpVerificationOnUpdate();
+        }
+
+        if (MapUtils.isEmpty(claims)) {
+            // Not required to handle in this handler.
+            Utils.setThreadLocalToSkipSendingSmsOtpVerificationOnUpdate(IdentityRecoveryConstants
+                    .SkipMobileNumberVerificationOnUpdateStates.SKIP_ON_INAPPLICABLE_CLAIMS.toString());
+            return;
+        }
+
+        String mobileNumber = claims.get(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM);
+
+        if (StringUtils.isNotBlank(mobileNumber)) {
+            String existingMobileNumber;
+            String username = user.getUserName();
+            try {
+                existingMobileNumber = userStoreManager.getUserClaimValue(username, IdentityRecoveryConstants.
+                        MOBILE_NUMBER_CLAIM, null);
+            } catch (UserStoreException e) {
+                String error = String.format("Error occurred while retrieving existing mobile number for user: %s in " +
+                        "domain : %s and user store : %s", username, user.getTenantDomain(), user.getUserStoreDomain());
```

Comment:
```suggestion
                        "domain: %s and user store: %s", username, user.getTenantDomain(), user.getUserStoreDomain());
```

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/1f1503b13f90b814eca821c6fca73b9af4c61052